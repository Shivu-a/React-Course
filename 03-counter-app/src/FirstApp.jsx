import PropTypes from "prop-types";

function App({ title, subTitle }) {
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <h2> {subTitle} </h2>
    </>
  );
}

App.propTypes = {
  title: PropTypes.string.isRequired,
};

App.defaultProps = {
  title: "No hay titulo",
  subTitle: "No hay sub-titulo",
};

export default App;
