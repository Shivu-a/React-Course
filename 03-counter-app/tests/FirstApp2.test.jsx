import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en FirstApp.jsx", () => {
  const titulo = "Hola! Soy Rimuru el slime!";
  const subTitulo = "No soy un slime malo!";

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={titulo} />);

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Hola! Soy Rimuru el slime!"', () => {
    render(<FirstApp title={titulo} />);
    expect(screen.getByText(titulo)).toBeTruthy;
  });

  test("Debe de mostrar el titulo en un H1", () => {
    render(<FirstApp title={titulo} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      titulo
    );
  });

  test("Debe de mostrar el subtitulo enviado por las props", () => {
    render(<FirstApp title={titulo} subTitle={subTitulo} />);

    expect(screen.getByText(subTitulo).innerHTML).toContain(subTitulo);
  });
});
