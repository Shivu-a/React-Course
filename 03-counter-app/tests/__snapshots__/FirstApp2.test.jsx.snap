// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp.jsx Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola! Soy Rimuru el slime!
     
  </h1>
  <h2>
     
    No hay sub-titulo
     
  </h2>
</div>
`;
